source/docs/casper/dapp-dev-guide/writing-contracts/contracts-and-session-code.md

```diff
@@ -0,0 +1,579 @@
+        Some(key) => key,
+        None => runtime::revert(ApiError::MissingKey)
+    }.into_uref().unwrap_or_revert();
+    system::transfer_from_purse_to_account(stored_purse_uref, caller, amount, None);
+}
+```
+To redeem from the Vault Contract (C2), call the "redeem" Entry Point of Vault Contract (C2).
+Above you see an example from [main.rs](https://github.com/jonas089/C3PRL0CK/blob/master/contract/src/main.rs) in Contract (C1).
+C1 and C2 have to share this entry point, as C2 is installed through C1.
+We are however not interested in calling any entry point other than "migrate" in the context of C1.
+You can use the casper-client or an SDK to call entry points on Smart Contracts.
+
+2. Approve Entry Point:
+
+```rust
+#[no_mangle]
+pub extern "C" fn approve(){
+    let caller: AccountHash = runtime::get_caller();
+    let owner_account_uref: URef = match runtime::get_key(OWNER_ACCOUNT){
+        Some(key) => key,
+        None => runtime::revert(ApiError::MissingKey)
+    }.into_uref().unwrap_or_revert();
+    let owner_account: AccountHash = storage::read_or_revert(owner_account_uref);
+    // only the owner of the contract instance is allowed to approve accounts for redemption.
+    if owner_account != caller{
+        runtime::revert(ApiError::PermissionDenied);
+    };
+    let new_account: AccountHash = runtime::get_named_arg(ARG_ACCOUNT);
+    let approved_list_uref: URef = match runtime::get_key(APPROVED_LIST){
+        Some(key) => key,
+        None => runtime::revert(ApiError::MissingKey)
+    }.into_uref().unwrap_or_revert();
+    let approved_list = storage::dictionary_get::<Vec<AccountHash>>(approved_list_uref, &owner_account.to_string()).unwrap_or_revert();
+    let res = match approved_list{
+        Some(mut v) => {
+            v.push(new_account);
+            v
+        },
+        None => {
+            let mut _approved_list: Vec<AccountHash> = Vec::new();
+            _approved_list.push(new_account);
+            _approved_list
+        }
+    };
+    storage::dictionary_put(approved_list_uref, &owner_account.to_string(), res);
+}
+```
+The "destination" purse is not the only named key that C2 holds. We also specified an approval list in C2's named keys.
+To successfully redeem funds from C2, one has to either be the installer/ "owner" or a member of the approval list.
+The installer can add account_hashes to the approval list through the "approve" entry point described above.
+
+## Videos and Tutorials
+
+To learn more about versioning contracts, consult the following video, which builds upon the previous topics and videos in the [Writing On-Chain Code](/writing-contracts) documentation.
+
+<p align="center">
+<iframe width="400" height="225" src="https://www.youtube.com/embed?v=sUg0nh3K3iQ&list=PL8oWxbJ-csEqi5FP87EJZViE2aLz6X1Mj&index=10" frameborder="0" allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</p>
+
```

Comment:
![image](https://user-images.githubusercontent.com/108756302/217269477-4ab6aa2a-67f3-45f9-8138-e8b872918ee1.png)

Reply:
<img width="888" alt="Screenshot 2023-02-07 at 15 49 55" src="https://user-images.githubusercontent.com/49498646/217278236-68e4b909-70bd-4937-ae73-5b20401b489a.png">
Example: width: 100%, height(fixed) 400px

Reply:
I think that looks much more usable.

Reply:
I agree that we need to standardize. I am seeing the same issue in other documentation. So, what i propose is that i approve this PR right now, and we create a separate "housekeeping" story to clean up these embeds.